packages/ocapn/src/client/index.js

```diff
@@ -0,0 +1,586 @@
+    getOutgoingConnection: locationId => {
+      const pendingSession = pendingSessions.get(locationId);
+      if (pendingSession === undefined) {
+        return undefined;
+      }
+      return pendingSession.outgoingConnection;
+    },
+    getSessionForConnection: connection => {
+      return connectionToSession.get(connection);
+    },
+    deleteConnection: connection => {
+      connectionToSession.delete(connection);
+    },
+    getPendingSessionPromise: locationId => {
+      const pendingSession = pendingSessions.get(locationId);
+      if (pendingSession === undefined) {
+        return undefined;
+      }
+      return pendingSession.promise;
+    },
+    resolveSession: (locationId, connection, session) => {
+      if (activeSessions.has(locationId)) {
+        throw Error(
+          `Unable to resolve session for ${locationId}. Active session already exists.`,
+        );
+      }
+      activeSessions.set(locationId, session);
+      connectionToSession.set(connection, session);
+      sessionIdToPeerPublicKey.set(toHex(session.id), session.peer.publicKey);
+      const pendingSession = pendingSessions.get(locationId);
+      if (pendingSession !== undefined) {
+        pendingSession.resolve(session);
+        pendingSessions.delete(locationId);
+      }
+    },
+    endSession: session => {
+      const locationId = locationToLocationId(session.peer.location);
+      const pendingSession = pendingSessions.get(locationId);
+      if (pendingSession !== undefined) {
+        pendingSession.reject(Error('Session ended.'));
+        pendingSessions.delete(locationId);
+      }
+      activeSessions.delete(locationId);
+      connectionToSession.delete(session.connection);
+    },
+    makePendingSession: (locationId, outgoingConnection) => {
+      if (activeSessions.has(locationId)) {
+        throw Error(
+          `Active session for location already exists: ${locationId}`,
+        );
+      }
+      if (pendingSessions.has(locationId)) {
+        throw Error(
+          `Pending session for location already exists: ${locationId}`,
+        );
+      }
+      const { promise, resolve, reject } = makePromiseKit();
+      /** @type {PendingSession} */
+      const pendingSession = harden({
+        outgoingConnection,
+        promise,
+        resolve,
+        reject,
+      });
+      pendingSessions.set(locationId, pendingSession);
+      return pendingSession;
+    },
+    getPeerPublicKeyForSessionId: sessionId => {
+      return sessionIdToPeerPublicKey.get(toHex(sessionId));
+    },
+  });
+};
+
+/**
+ * @param {object} [options]
+ * @param {string} [options.debugLabel]
+ * @param {boolean} [options.verbose]
+ * @param {Map<string, any>} [options.swissnumTable]
+ * @param {Map<string, any>} [options.giftTable]
+ * @returns {Client}
+ */
+export const makeClient = ({
+  debugLabel = 'ocapn',
+  verbose = false,
+  swissnumTable = new Map(),
+  giftTable = new Map(),
```

Comment:
Given that the caller can retain these options, is there an expectation that they can update them or observe changes to them after calling `makeClient`? If there is no such expectation, we should capture a snapshot and discard the option. If there is such an expectation, I’m weary about the semantics and the observability story.